### PR TITLE
feature: Enhance Search bar with query parameter

### PR DESCRIPTION
**Pull request by JimTheCat**
- Add SearchResults which will react with query parameter,
- Enhance Search mechanizm.

Refs: 8697653ha

### CODE REVIEW

**Comment by JimTheCat** [comment]
Task linked: [CU-8697653ha Enhance Search bar with query parameter](https://app.clickup.com/t/8697653ha)

**Comment by JimTheCat** [comment]
image
image
image
image
Small disclaimer. Results of searching is static so it doesn't filter the results. That will be a part of backend to return already filtered data.

File: frontend/src/Features/Search/Search.tsx [hunk 1]
```diff
@@ -1,93 +1,148 @@
...
+    const filteredOptions = (options as ComboboxItem[]).filter((option) => {
       const searchLower = search.toLowerCase();
-      return option.value.toLowerCase().includes(searchLower) || usersData[option.value].tag.toLowerCase().includes(searchLower);
+      return option.value.toLowerCase().includes(searchLower) || usersData[option.value]?.tag.toLowerCase().includes(searchLower);
     });
+
+    // Add dynamic option for searching users
+    return [
+      {
+        value: `search:${search}`,
```
Comment (KinTrae) [inline frontend/src/Features/Search/Search.tsx:85]: One starts with capital letter, other with lowercase. Is it correct?

Comment (JimTheCat) [reply frontend/src/Features/Search/Search.tsx:85]: Yup it is. search: is used for query operations and this "Wyszukaj" is just a label which is visible during searching for users.

Comment (KinTrae) [reply frontend/src/Features/Search/Search.tsx:85]: okie, no more questions

**Review by KinTrae — APPROVED** [review]

**Pull request merged**